test/requirements.txt

```diff
@@ -32,77 +39,155 @@ certifi==2022.6.15
     #   requests
 charset-normalizer==2.1.0
     # via requests
+click==8.1.7
+    # via
+    #   typer
+    #   uvicorn
+colorama==0.4.6
+    # via typer
+contourpy==1.2.0
+    # via matplotlib
+coverage[toml]==7.4.1
+    # via
+    #   -r requirements.in
+    #   pytest-cov
+cycler==0.12.1
+    # via matplotlib
 decorator==5.1.1
     # via ipython
+diffusers==0.23.1
+    # via -r requirements.in
+dnspython==2.5.0
+    # via email-validator
+email-validator==2.1.0.post1
+    # via pydantic
 entrypoints==0.4
     # via altair
+fastapi==0.103.0
+    # via
+    #   -r requirements.in
+    #   gradio
+ffmpy==0.3.1
+    # via gradio
 filelock==3.7.1
     # via
+    #   diffusers
     #   huggingface-hub
+    #   torch
     #   transformers
-gradio_pdf==0.0.3
+fonttools==4.48.1
+    # via matplotlib
+fsspec==2024.2.0
+    # via
+    #   gradio-client
+    #   huggingface-hub
+    #   torch
+gradio==4.18.0
+    # via gradio-pdf
```

Comment:
Gradio itself shouldn't be a test requirement...

Reply:
Makes sense, I have moved it out of the if block.

Reply:
I have removed this from the file, but I may need your help here as i am using pip-compile for the first time and it added a lot of dependencies in this file. Can you please take a look if it is correct or not.